From: Tim Gardner <tim.gardner@canonical.com>
Date: Thu, 25 Sep 2008 07:09:25 -0600
Subject: Disable the fallback clause in hwclock when /dev/rtc cannot be
 opened.

hwclock should not access the x86 RTC using I/O instructions unless explicitly
requested from the command line (--directisa). This issue was encountered
while debugging a boot time race when hwclock.sh was run in parallel.

Addresses-Ubuntu-Bug: 274402
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
Signed-off-by: LaMont Jones <lamont@debian.org>

Conflicts:
	hwclock/hwclock.c
---
 sys-utils/hwclock.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/sys-utils/hwclock.c
+++ b/sys-utils/hwclock.c
@@ -1226,8 +1226,18 @@
 {
 	ur = NULL;
 
+  /*
+   * This final clause is a really bad idea on x86/AT PCs. You run the
+   * risk of a race condition with another copy of hwclock
+   * that already has /dev/rtc open. The fallback case on
+   * x86 is to then raise I/O priviledge level and access
+   * the RTC CMOS directly using I/O instructions. Simultaneous
+   * access like that can really hose the RTC.
+   */
+#if !defined(__i386__)
 	if (user_requests_ISA)
 		ur = probe_for_cmos_clock();
+#endif
 
 #ifdef __linux__
 	if (!ur)
